src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file PorousSolid.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_POROUSSOLID_HPP_
+#define GEOSX_CONSTITUTIVE_SOLID_POROUSSOLID_HPP_
+
+#include "constitutive/solid/CoupledSolid.hpp"
+#include "constitutive/solid/porosity/BiotPorosity.hpp"
+#include "constitutive/solid/SolidBase.hpp"
+#include "constitutive/permeability/ConstantPermeability.hpp"
+
+namespace geosx
+{
+namespace constitutive
+{
+
+/**
+ * @brief Provides kernel-callable constitutive update routines
+ *
+ *
+ * @tparam SOLID_TYPE
+ */
+template< typename SOLID_TYPE >
+class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity, ConstantPermeability >
+{
+public:
+
+  using DiscretizationOps = typename SOLID_TYPE::KernelWrapper::DiscretizationOps;
+
+  /**
+   * @brief Constructor
+   */
+  PorousSolidUpdates( SOLID_TYPE const & solidModel,
+                      BiotPorosity const & porosityModel,
+                      ConstantPermeability const & permModel ):
+    CoupledSolidUpdates< SOLID_TYPE, BiotPorosity, ConstantPermeability >( solidModel, porosityModel, permModel )
+  {}
+
+  GEOSX_HOST_DEVICE
+  void smallStrainUpdate( localIndex const k,
+                          localIndex const q,
+                          real64 const & deltaPressure,
+                          real64 const ( &strainIncrement )[6],
+                          real64 ( & stress )[6],
+                          real64 & dPorosity_dPressure,
+                          real64 & dPorosity_dVolStrain,
+                          real64 & dTotalStress_dPressure,
+                          DiscretizationOps & stiffness ) const
+  {
+    m_solidUpdate.smallStrainUpdate( k, q, strainIncrement, stress, stiffness );
+
+    updateBiotCoefficient( k );
```

Comment:
Using stiffness in a follow on PR..correct?

Reply:
yes